### PR TITLE
Add lightning profiler

### PR DESCRIPTION
**Pull request by porterrf**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs?
- [-] Did you write any tests to validate this change?  
- [-] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

Allows user to pass in PyTorch Lightning profiler to the PL estimator

## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

File: examples/spark/pytorch/pytorch_lightning_spark_mnist.py [hunk 1]
```diff
@@ -184,6 +188,14 @@ def on_train_end(self, trainer, model):
                                    verbose=True,
                                    mode='max'))
 
+    if False:
```
Comment (chongxiaoc) [inline examples/spark/pytorch/pytorch_lightning_spark_mnist.py:191]: What is the reason of hard-coding `False` here?

File: examples/spark/pytorch/pytorch_lightning_spark_mnist.py [hunk 2]
```diff
@@ -184,6 +188,14 @@ def on_train_end(self, trainer, model):
...
+        prof_dir = tempfile.mkdtemp()
```
Comment (chongxiaoc) [inline examples/spark/pytorch/pytorch_lightning_spark_mnist.py:192]: Maybe we should just use `log_dir`?

**Review by chongxiaoc — COMMENTED** [review]
Some comments about log_dir location.

File: examples/spark/pytorch/pytorch_lightning_spark_mnist.py (hunk 1 above)
Comment (porterrf) [reply examples/spark/pytorch/pytorch_lightning_spark_mnist.py]: That was a miss from testing, fixed

File: examples/spark/pytorch/pytorch_lightning_spark_mnist.py (hunk 2 above)
Comment (porterrf) [reply examples/spark/pytorch/pytorch_lightning_spark_mnist.py]: Yes, makes sense.  When we don't pass in a dirname, lightning will pick the dirname from the logger's log_path.

File: horovod/spark/lightning/remote.py [hunk 3]
```diff
@@ -190,6 +200,7 @@ def train(serialized_model):
 
             torch.save(output, serialized_checkpoint)
             serialized_checkpoint.seek(0)
+
```
Comment (chongxiaoc) [inline horovod/spark/lightning/remote.py:203]: remove empty line.

**Review by chongxiaoc — CHANGES_REQUESTED** [review]
Add `PyTorchProfiler()`into one of the unit test of lightning estimator.

**Review by chongxiaoc — APPROVED** [review]
LGTM

**Review by chongxiaoc — CHANGES_REQUESTED** [review]
Need one more fix for pytorch_1_6_0 failure for example.

**Review by chongxiaoc — APPROVED** [review]

File: horovod/spark/lightning/remote.py [hunk 4]
```diff
@@ -106,6 +106,9 @@ def train(serialized_model):
 
             if train_logger is None:
                 train_logger = TensorBoardLogger(logs_path)
+            elif isinstance(train_logger, CometLogger) and train_logger._save_dir is None:
+                # Setting the CometLogger's save_dir allows us to sync checkpoints and profiler output
+                train_logger._save_dir = logs_path
```
Comment (chongxiaoc) [inline horovod/spark/lightning/remote.py:111]: great

**Review by irasit — APPROVED** [review]

**Pull request merged**